src/components/organisms/PageHeader/ClusterSelection.tsx

```diff
@@ -150,61 +134,117 @@ const ClusterSelection = ({previewResource}: {previewResource?: K8sResource}) =>
-                <S.ClusterAccessContainer>{icon}</S.ClusterAccessContainer>
-              </Tooltip>
+              <>
+                <Tooltip mouseEnterDelay={TOOLTIP_DELAY} title={tooltip}>
+                  <S.ClusterAccessContainer>{icon}</S.ClusterAccessContainer>
+                </Tooltip>
+                <S.ClusterOutlined />
+                <Dropdown
+                  overlay={<ClusterSelectionTable setIsClusterDropdownOpen={setIsClusterDropdownOpen} />}
+                  overlayClassName="cluster-dropdown-item"
+                  placement="bottom"
+                  arrow
+                  trigger={['click']}
+                  disabled={previewLoader.isLoading || isInPreviewMode}
+                  visible={isClusterDropdownOpen}
+                  onVisibleChange={setIsClusterDropdownOpen}
+                >
+                  <S.ClusterButton type="link" ref={dropdownButtonRef} size="small">
+                    <S.ClusterContextName>{kubeConfigContext}</S.ClusterContextName>
+                    <S.DownOutlined />
+                  </S.ClusterButton>
+                </Dropdown>
+              </>
             )}
-            <span>{isKubeConfigPathValid ? 'Configured' : 'No Cluster Configured'}</span>
-          </S.ClusterStatusText>
-
-          {isKubeConfigPathValid ? (
-            <>
-              <S.Divider type="vertical" />
-              <Tooltip mouseEnterDelay={TOOLTIP_DELAY} mouseLeaveDelay={0} title={ClusterModeTooltip} placement="right">
-                <S.Button type="link" disabled={isAccessLoading} onClick={handleLoadCluster}>
-                  {createClusterObjectsLabel()}
-                </S.Button>
-              </Tooltip>
-            </>
-          ) : (
-            <>
+            <S.ClusterStatusText
+              isKubeConfigPathValid={isKubeConfigPathValid}
+              isInPreviewMode={!previewLoader.isLoading && isInPreviewMode}
+              previewType={previewType}
+            >
+              {isKubeConfigPathValid ? (
+                <S.CheckCircleOutlined
+                  isKubeConfigPathValid={isKubeConfigPathValid}
+                  isInPreviewMode={!previewLoader.isLoading && isInPreviewMode}
+                  previewType={previewType}
+                />
+              ) : (
+                <S.ClusterOutlined />
+              )}
+              <span>{isKubeConfigPathValid ? 'Configured' : 'NO CLUSTER CONFIGURED'}</span>
+            </S.ClusterStatusText>
+
+            {!isKubeConfigPathValid && (
               <S.ClusterActionButton type="link" onClick={handleClusterConfigure}>
                 Configure
               </S.ClusterActionButton>
-            </>
-          )}
-        </S.ClusterStatus>
+            )}
+          </S.ClusterStatus>
+        </>
       )}
+      <>
+        {isKubeConfigPathValid && activeProject && (
+          // <Tooltip mouseEnterDelay={TOOLTIP_DELAY} mouseLeaveDelay={0} title={ClusterModeTooltip} placement="right">
```

Comment:
commented code

Reply:
@erdkse good work! I agree with Wito that the dropdown could benefit from some padding on the right for the longest item. Same for the "+" dropdown, you can add some padding on the right when displayed, so "new form local folder" has a bit more air to the right.

Reply:
Ok thank you folks for the reviews, I made fixes according to them. Now I am merging and I can add further wishes on following PRs